com.unity.multiplayer.mlapi/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -458,13 +457,13 @@ private void InjectWriteAndCallBlocks(MethodDefinition methodDefinition, CustomA
             var instructions = new List<Instruction>();
             var processor = methodDefinition.Body.GetILProcessor();
             var isServerRpc = rpcAttribute.AttributeType.FullName == CodeGenHelpers.ServerRpcAttribute_FullName;
-            var isReliableRpc = true;
+            var rpcDelivery = RpcDelivery.Reliable;
             foreach (var attrField in rpcAttribute.Fields)
             {
                 switch (attrField.Name)
                 {
-                    case nameof(RpcAttribute.IsReliable):
-                        isReliableRpc = attrField.Argument.Type == typeSystem.Boolean && (bool)attrField.Argument.Value;
+                    case nameof(RpcAttribute.Delivery):
+                        rpcDelivery = (RpcDelivery)attrField.Argument.Value;
```

Comment:
I don't like this aspect of the C# language.  I've run into this where even if I define the enum as a byte, I have to do these nasty casts everywhere.

Reply:
yeah, in C/C++ enums are just named integers but C# treats them as standalone types, so...